community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/Foldable.scala

```diff
@@ -92,6 +95,18 @@ object Foldable {
           existsAcc(remaining ++= that.reverseChildren, f)
       }
     }
+
+  @tailrec
+  private def findAcc[A: Manifest](remaining: mutable.ArrayStack[Any]): A =
```

Comment:
Surprised to see this change at all, need to trace back what happened here. The use of manifests is all over rewriter logic I fear. We should change that one day :) (Perhaps after 2.11?)